lib/std/linked_list.zig

```diff
@@ -92,14 +92,20 @@ pub fn SinglyLinkedList(comptime T: type) type {
         ///
         /// Arguments:
         ///     node: Pointer to the node to be removed.
-        pub fn remove(list: *Self, node: *Node) void {
+        pub fn remove(list: *Self, node: *Node) !void {
+            if (list.first == null) {
+                return error.EmptyList;
+            }
             if (list.first == node) {
                 list.first = node.next;
             } else {
                 var current_elm = list.first.?;
-                while (current_elm.next != node) {
+                while ((current_elm.next != null) and (current_elm.next != node)) {
                     current_elm = current_elm.next.?;
                 }
+                if (current_elm.next == null) {
+                    return error.TargetNodeNotInList;
+                }
                 current_elm.next = node.next;
             }
         }
```

Comment:
Mostly a style/personal preference thing, but I'd write this as:

```zig
        pub fn remove(list: *Self, node: *Node) !void {
            if (list.first == node) {
                list.first = node.next;
            } else if (list.first != null) {
                var current_elm = list.first.?;
                while (current_elm.next != node) {
                    if (current_elm.next == null) return error.TargetNodeNotInList;
                    current_elm = current_elm.next.?;
                }
                current_elm.next = node.next;
            }
            return error.EmptyList;
        }
```

or with it returning a bool:

```zig
        pub fn remove(list: *Self, node: *Node) bool {
            if (list.first == node) {
                list.first = node.next;
            } else if (list.first != null) {
                var current_elm = list.first.?;
                while (current_elm.next != node) {
                    if (current_elm.next == null) return false;
                    current_elm = current_elm.next.?;
                }
                current_elm.next = node.next;
                return true;
            }
            return false;
        }
```

Reply:
thanks